resources/aws_ec2.go

```diff
@@ -379,174 +388,101 @@ func (obj *AwsEc2Res) Watch() error {
 
 // longpollWatch uses the ec2 api's built in methods to watch ec2 resource state.
 func (obj *AwsEc2Res) longpollWatch() error {
-	send := false
+	var send bool
 	var exit *error
+
+	// we tell the engine we're running right away. Because watchWaiter
+	// checks the instance state directly so we don't miss any events.
 	if err := obj.Running(); err != nil {
 		return err
 	}
-	defer obj.wg.Wait()
-	defer close(obj.closeChan)
+
+	// channels and waitgroup for goroutines.
```

Comment:
I forgot to change these back to the struct channels and wg. I can change it back after you review.